### PR TITLE
Use array_* functions where possible 

### PR DESCRIPTION
**Pull request by c-harris**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ script:
   - mkdir -p build/logs
   - vendor/bin/phpunit --coverage-clover build/logs/clover.xml
 
-after_script:
+after_sucess:
```
Comment (CyberiaResurrection) [inline .travis.yml]: Typo?

Comment (c-harris) [reply .travis.yml]: Correct

**Comment by c-harris** [comment]
@CyberiaResurrection this is ready, could you please review?

File: src/ClassFinder.php [hunk 2]
```diff
@@ -42,18 +42,10 @@ private static function getProjectClasses(string $namespace): array
...
-        }, ARRAY_FILTER_USE_BOTH);
-        // now class list of maps are assembled, use class_exists calls to explicitly autoload them,
-        // while not running them
-        foreach ($map as $class => $file) {
-            class_exists($class, true);
-        }
-        self::$loadedNamespaces[] = $namespace;
+        array_walk($map, function ($filename, $className, $namespace) {
+            assert(file_exists($filename), $filename);
+            self::strStartsWith($namespace, $className) && class_exists($className);
```
Comment (CyberiaResurrection) [inline src/ClassFinder.php]: The original call to class_exists also passed true explicitly - maybe restore that for readability?

File: src/ClassFinder.php [hunk 3]
```diff
@@ -153,12 +153,9 @@ public static function getClasses(string $namespace = '', callable $conditional
             $conditional,
             $includeVendor
         ) {
-            /*$dontSkip = true;
-            if (!$includeVendor) {
-                $dontSkip = !self::isClassInVendor($class);
-            }*/
-            $dontSkip = $includeVendor || !self::isClassInVendor($class);
-            return substr($class, 0, strlen($namespace)) === $namespace && $dontSkip && $conditional($class) ;
+            return self::strStartsWith($namespace, $class) &&
```
Comment (CyberiaResurrection) [inline src/ClassFinder.php]: Is this strStartsWith check added to make it stone obvious we're only wanting things in $namespace?

File: src/ClassFinder.php [hunk 4]
```diff
@@ -71,13 +71,10 @@ private static function getClassMap(string $namespace): array
...
-        $projectDirs = self::getProjectSearchDirs($namespace);
-        $map = [];
-        // Use composer's ClassMapGenerator to pull the class list out of each project search directory
-        foreach ($projectDirs as $dir) {
-            $map = array_merge($map, ClassMapGenerator::createMap($dir));
-        }
-        return $map;
+        return array_reduce(self::getProjectSearchDirs($namespace),
+            function($map, $dir) {
+                return array_merge($map, ClassMapGenerator::createMap($dir));
```
Comment (CyberiaResurrection) [inline src/ClassFinder.php]: Restore the removed comment to make it stone obvious to lesser minds than you what the blazes you're doing?

File: src/ClassFinder.php (hunk 3 above)
Comment (c-harris) [reply src/ClassFinder.php]: Get project classes autoload in the namespace is then return ALL declared classes (so we can cache the loaded namespace)

We need to filter the result of get_declared_classes

File: src/ClassFinder.php (hunk 4 above)
Comment (c-harris) [reply src/ClassFinder.php]: Done

File: src/ClassFinder.php (hunk 2 above)
Comment (c-harris) [reply src/ClassFinder.php]: Done

**Pull request closed without merging**